App crash at Create new wallet

**steps:**
docker stop fulmine
docker start fulmine
tap Forgot Password?
tap Create new wallet

**Expected results:**
Able to start the new wallet creation process

**Actual results:**
App crashes

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xc07d41]
```

Video capture: 
https://github.com/user-attachments/assets/d6f779a2-eb3a-4129-998b-a7c33ccd0039

"version":"latest","commit":"f16103d","date":"2025-04-01T13:10:07Z"

--- a/cmd/fulmine/main.go
+++ b/cmd/fulmine/main.go
@@ -46,11 +46,12 @@ func main() {
 		WithTLS:  cfg.WithTLS,
 	}
 
-	storeSvc, err := store.NewStore(store.Config{
+	storeCfg := store.Config{
 		BaseDir:          cfg.Datadir,
 		ConfigStoreType:  configStoreType,
 		AppDataStoreType: appDataStoreType,
-	})
+	}
+	storeSvc, err := store.NewStore(storeCfg)
 	if err != nil {
 		log.WithError(err).Fatal(err)
 	}
@@ -81,6 +82,7 @@ func main() {
 
 	appSvc, err := application.NewService(
 		buildInfo,
+		storeCfg,
 		storeSvc,
 		settingsRepo,
 		vhtlcRepo,

--- a/internal/core/application/service.go
+++ b/internal/core/application/service.go
@@ -19,6 +19,7 @@ import (
 	arksdk "github.com/ark-network/ark/pkg/client-sdk"
 	"github.com/ark-network/ark/pkg/client-sdk/client"
 	grpcclient "github.com/ark-network/ark/pkg/client-sdk/client/grpc"
+	"github.com/ark-network/ark/pkg/client-sdk/store"
 	"github.com/ark-network/ark/pkg/client-sdk/types"
 	"github.com/btcsuite/btcd/btcec/v2"
 	"github.com/btcsuite/btcd/btcec/v2/schnorr"
@@ -49,6 +50,7 @@ type Service struct {
 	BuildInfo BuildInfo
 
 	arksdk.ArkClient
+	storeCfg         store.Config
 	storeRepo        types.Store
 	settingsRepo     domain.SettingsRepository
 	vhtlcRepo        domain.VHTLCRepository
@@ -81,6 +83,7 @@ type Notification struct {
 
 func NewService(
 	buildInfo BuildInfo,
+	storeCfg store.Config,
 	storeSvc types.Store,
 	settingsRepo domain.SettingsRepository,
 	vhtlcRepo domain.VHTLCRepository,
@@ -101,6 +104,7 @@ func NewService(
 		svc := &Service{
 			BuildInfo:        buildInfo,
 			ArkClient:        arkClient,
+			storeCfg:         storeCfg,
 			storeRepo:        storeSvc,
 			settingsRepo:     settingsRepo,
 			vhtlcRepo:        vhtlcRepo,
@@ -136,6 +140,7 @@ func NewService(
 	svc := &Service{
 		BuildInfo:        buildInfo,
 		ArkClient:        arkClient,
+		storeCfg:         storeCfg,
 		storeRepo:        storeSvc,
 		settingsRepo:     settingsRepo,
 		vhtlcRepo:        vhtlcRepo,
@@ -289,10 +294,6 @@ func (s *Service) UnlockNode(ctx context.Context, password string) error {
 }
 
 func (s *Service) Reset(ctx context.Context) error {
-	if err := s.isInitializedAndUnlocked(ctx); err != nil {
-		return err
-	}
-
 	backup, err := s.settingsRepo.GetSettings(ctx)
 	if err != nil {
 		return err
@@ -305,6 +306,17 @@ func (s *Service) Reset(ctx context.Context) error {
 		s.settingsRepo.AddSettings(ctx, *backup)
 		return err
 	}
+
+	s.Stop()
+	// nolint:all
+	storeSvc, _ := store.NewStore(s.storeCfg)
+	// nolint:all
+	cli, _ := arksdk.NewCovenantlessClient(storeSvc)
+	// TODO: Maybe drop?
+	// nolint:all
+	s.settingsRepo.AddDefaultSettings(ctx)
+	s.ArkClient = cli
+	s.storeRepo = storeSvc
 	return nil
 }
 
@@ -370,10 +382,6 @@ func (s *Service) GetTotalBalance(ctx context.Context) (uint64, error) {
 }
 
 func (s *Service) GetRound(ctx context.Context, roundId string) (*client.Round, error) {
-	if err := s.isInitializedAndUnlocked(ctx); err != nil {
-		return nil, err
-	}
-
 	return s.grpcClient.GetRoundByID(ctx, roundId)
 }
 

--- a/internal/interface/web/api.go
+++ b/internal/interface/web/api.go
@@ -85,6 +85,15 @@ func (s *service) disconnectLNDApi(c *gin.Context) {
 	reload(c)
 }
 
+func (s *service) forgotApi(c *gin.Context) {
+	if err := s.svc.Reset(c); err != nil {
+		toast := components.Toast("Unable to delete previous wallet", true)
+		toastHandler(toast, c)
+		return
+	}
+	c.Redirect(http.StatusFound, "/welcome")
+}
+
 func (s *service) validateNoteApi(c *gin.Context) {
 	var data gin.H
 	note := c.PostForm("note")

--- a/internal/interface/web/api.go
+++ b/internal/interface/web/api.go
@@ -91,7 +91,7 @@ func (s *service) forgotApi(c *gin.Context) {
 		toastHandler(toast, c)
 		return
 	}
-	c.Redirect(http.StatusFound, "/welcome")
+	redirect("/welcome", c)
 }
 
 func (s *service) validateNoteApi(c *gin.Context) {

--- a/internal/interface/web/handlers.go
+++ b/internal/interface/web/handlers.go
@@ -187,7 +187,6 @@ func (s *service) lock(c *gin.Context) {
 }
 
 func (s *service) unlock(c *gin.Context) {
-	log.Infof("referer %s", c.Request.Referer())
 	bodyContent := pages.Unlock()
 	s.pageViewHandler(bodyContent, c)
 }

--- a/internal/interface/web/handlers.go
+++ b/internal/interface/web/handlers.go
@@ -109,15 +109,6 @@ func (s *service) events(c *gin.Context) {
 	}
 }
 
-func (s *service) forgot(c *gin.Context) {
-	if err := s.svc.Reset(c); err != nil {
-		toast := components.Toast("Unable to delete previous wallet", true)
-		toastHandler(toast, c)
-		return
-	}
-	c.Redirect(http.StatusFound, "/welcome")
-}
-
 func (s *service) index(c *gin.Context) {
 	bodyContent := pages.Welcome()
 	if s.svc.IsReady() {

--- a/internal/interface/web/server.go
+++ b/internal/interface/web/server.go
@@ -79,7 +79,6 @@ func NewService(appSvc *application.Service, stopCh chan struct{}) *service {
 	svc.GET("/backup/ack", svc.backupAck)
 	svc.GET("/backup/tab/:active", svc.backupTabActive)
 	svc.GET("/done", svc.done)
-	svc.GET("/forgot", svc.forgot)
 	svc.GET("/hero", svc.getHero)
 	svc.GET("/import", svc.importWalletPrivateKey)
 	svc.GET("/lock", svc.lock)
@@ -115,6 +114,7 @@ func NewService(appSvc *application.Service, stopCh chan struct{}) *service {
 	svc.POST("/swap/confirm", svc.swapConfirm)
 
 	svc.POST("/helpers/claim/:txid", svc.claimTx)
+	svc.POST("/helpers/forgot", svc.forgotApi)
 	svc.POST("/helpers/lock", svc.lockApi)
 	svc.POST("/helpers/settings", svc.updateSettingsApi)
 	svc.POST("/helpers/node/connect", svc.connectLNDApi)

--- a/internal/interface/web/templates/pages/locked.templ
+++ b/internal/interface/web/templates/pages/locked.templ
@@ -5,6 +5,7 @@ import (
 )
 
 templ lockPage(kind string) {
+  <form hx-post="/helpers/forgot" id="forgot"><button class="hidden" type="submit" /></form>
   <form hx-post={"/helpers/" + kind}>
     <div class="bg-radial-gradient h-screen fullscreen">
       <div class="flex flex-col h-screen justify-between md:h-auto md:gap-10 w-80 m-auto p-3">
@@ -28,7 +29,7 @@ templ lockPage(kind string) {
               </p>
             </div>
             if kind == "unlock" {
-              <p class="text-right mt-2 cursor-pointer" hx-on:click="redirect('/forgot')">Forgot password?</p>
+              <p class="text-right mt-2 cursor-pointer" onclick="forgotPass()">Forgot password?</p>
             }
           </div>
         </div>
@@ -37,6 +38,11 @@ templ lockPage(kind string) {
         </button>
       </div>
     </div>
+    <script>
+      const forgotPass = () => {
+        document.querySelector("#forgot").querySelector("button").click()
+      }
+    </script>
   </form>
 }
 

--- a/internal/interface/web/templates/pages/locked_templ.go
+++ b/internal/interface/web/templates/pages/locked_templ.go
@@ -33,14 +33,14 @@ func lockPage(kind string) templ.Component {
 			templ_7745c5c3_Var1 = templ.NopComponent
 		}
 		ctx = templ.ClearChildren(ctx)
-		templ_7745c5c3_Err = templruntime.WriteString(templ_7745c5c3_Buffer, 1, "<form hx-post=\"")
+		templ_7745c5c3_Err = templruntime.WriteString(templ_7745c5c3_Buffer, 1, "<form hx-post=\"/helpers/forgot\" id=\"forgot\"><button class=\"hidden\" type=\"submit\"></button></form><form hx-post=\"")
 		if templ_7745c5c3_Err != nil {
 			return templ_7745c5c3_Err
 		}
 		var templ_7745c5c3_Var2 string
 		templ_7745c5c3_Var2, templ_7745c5c3_Err = templ.JoinStringErrs("/helpers/" + kind)
 		if templ_7745c5c3_Err != nil {
-			return templ.Error{Err: templ_7745c5c3_Err, FileName: `internal/interface/web/templates/pages/locked.templ`, Line: 8, Col: 35}
+			return templ.Error{Err: templ_7745c5c3_Err, FileName: `internal/interface/web/templates/pages/locked.templ`, Line: 9, Col: 35}
 		}
 		_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(templ.EscapeString(templ_7745c5c3_Var2))
 		if templ_7745c5c3_Err != nil {
@@ -83,7 +83,7 @@ func lockPage(kind string) templ.Component {
 			return templ_7745c5c3_Err
 		}
 		if kind == "unlock" {
-			templ_7745c5c3_Err = templruntime.WriteString(templ_7745c5c3_Buffer, 7, "<p class=\"text-right mt-2 cursor-pointer\" hx-on:click=\"redirect(&#39;/forgot&#39;)\">Forgot password?</p>")
+			templ_7745c5c3_Err = templruntime.WriteString(templ_7745c5c3_Buffer, 7, "<p class=\"text-right mt-2 cursor-pointer\" onclick=\"forgotPass()\">Forgot password?</p>")
 			if templ_7745c5c3_Err != nil {
 				return templ_7745c5c3_Err
 			}
@@ -95,13 +95,13 @@ func lockPage(kind string) templ.Component {
 		var templ_7745c5c3_Var3 string
 		templ_7745c5c3_Var3, templ_7745c5c3_Err = templ.JoinStringErrs(kind)
 		if templ_7745c5c3_Err != nil {
-			return templ.Error{Err: templ_7745c5c3_Err, FileName: `internal/interface/web/templates/pages/locked.templ`, Line: 36, Col: 15}
+			return templ.Error{Err: templ_7745c5c3_Err, FileName: `internal/interface/web/templates/pages/locked.templ`, Line: 37, Col: 15}
 		}
 		_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(templ.EscapeString(templ_7745c5c3_Var3))
 		if templ_7745c5c3_Err != nil {
 			return templ_7745c5c3_Err
 		}
-		templ_7745c5c3_Err = templruntime.WriteString(templ_7745c5c3_Buffer, 9, "</button></div></div></form>")
+		templ_7745c5c3_Err = templruntime.WriteString(templ_7745c5c3_Buffer, 9, "</button></div></div><script>\n      const forgotPass = () => {\n        document.querySelector(\"#forgot\").querySelector(\"button\").click()\n      }\n    </script></form>")
 		if templ_7745c5c3_Err != nil {
 			return templ_7745c5c3_Err
 		}

